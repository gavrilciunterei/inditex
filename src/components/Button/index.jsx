import React from 'react';

function Button({ text, handleOnClick, disabled }) {
  return (
    <button
      disabled={disabled}
      onClick={handleOnClick}
      className="disabled:border-0 disabled:bg-gray-400 disabled:hover:scale-100 disabled:hover:text-black disabled:text-black text-base undefined hover:scale-110 focus:outline-none flex justify-center px-4 py-2 rounded font-bold cursor-pointer hover:bg-teal-700 hover:text-teal-100 bg-teal-100 text-teal-700 border duration-200 ease-in-out border-teal-600 transition "
    >
      {text}
    </button>
  );
}

export default Button;
