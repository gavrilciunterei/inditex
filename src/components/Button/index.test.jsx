import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

it('Test Button', () => {
  const tree = renderer
    .create(<Button text="Texto" handleOnClick={true} disabled={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
