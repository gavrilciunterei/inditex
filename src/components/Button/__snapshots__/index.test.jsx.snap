// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button 1`] = `
<button
  className="disabled:border-0 disabled:bg-gray-400 disabled:hover:scale-100 disabled:hover:text-black disabled:text-black text-base undefined hover:scale-110 focus:outline-none flex justify-center px-4 py-2 rounded font-bold cursor-pointer hover:bg-teal-700 hover:text-teal-100 bg-teal-100 text-teal-700 border duration-200 ease-in-out border-teal-600 transition "
  disabled={false}
  onClick={true}
>
  Texto
</button>
`;
